tester/device8.go

```diff
@@ -55,8 +58,18 @@ func (d *I2CDevice8) WriteRegister(r uint8, buf []byte) error {
 
 // Tx implements I2C.Tx.
 func (bus *I2CDevice8) Tx(w, r []byte) error {
-	// TODO: implement this
-	return nil
+	switch len(w) {
+	case 0:
+		bus.c.Fatalf("i2c mock: need a write byte")
+		return nil
+	case 1:
+		return bus.ReadRegister(w[0], r)
+	default:
+		if len(r) > 0 || len(w) == 1 {
+			bus.c.Fatalf("i2c mock: unsupported lengths in Tx(%d, %d)", len(w), len(r))
+		}
+		return bus.WriteRegister(w[0], w[1:])
```

Comment:
I guess you could make `ReadRegister` and `WriteRegister` unexported now (meaning, `readRegister` and `writeRegister`) - but it's fine as-is.

Reply:
Fixed!